server/apps/research/serializers/article_serializer.py

```
@@ -9,73 +9,127 @@
+            # Bulk create related articles
+            related_articles = [
+                RelatedArticle(from_article=article, to_article=related_article)
+                for related_article in related_article_ids
+            ]
+            RelatedArticle.objects.bulk_create(related_articles)
+
             return article
-        except Exception as e:            
+
+        except Exception as e:
             raise serializers.ValidationError(f"Error creating article: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)